packages/framework/presence/src/test/notificationsManager.spec.ts

```diff
@@ -3,86 +3,480 @@
-		}>({
-			msg: (client: ISessionClient, message: string) => {
-				console.log(`${client.sessionId} says, "${message}"`);
-			},
-		}),
-	});
-	// Workaround ts(2775): Assertions require every name in the call target to be declared with an explicit type annotation.
-	const notifications: typeof notificationsWorkspace = notificationsWorkspace;
-
-	// TODO: inferences for Notifications additions are not working.
-	// They allow incorrect listener signatures and no named events
-	// produced after `add`.
-	const NF = Notifications({
-		newId: (client: ISessionClient, id: number): void => {
-			console.log(`${client.sessionId} has a new id: ${id}`);
-		},
-	});
+		beforeEach(() => {
+			logger = new EventAndErrorTrackingLogger();
+			runtime = new MockEphemeralRuntime(logger);
 
-	notifications.add("my_events", NF);
-	// Below explicit generic specifaction should not be required.
-	// 	Notifications<
-	// 		{
-	// 			newId: (id: number) => void;
-	// 		},
-	// 		"events"
-	// 	>({
-	// 		newId: (client: ISessionClient, id: number) => {
-	// 			console.log(`${client.sessionId} has a new id: ${id}`);
-	// 		},
-	// 	}),
-	// );
-
-	// "newId" should be allowed as a named event.
-	// map.my_events.emit.broadcast("newId", 42);
-
-	function logUnattended(name: string, client: ISessionClient, ...content: unknown[]): void {
-		console.log(
-			`${client.sessionId} sent unattended notification '${name}' with content`,
-			...content,
-		);
-	}
-
-	const chat = notifications.props.chat;
-
-	chat.emit.broadcast("msg", "howdy");
-
-	// Track clients that have started chatting
-	const chatClients = new Set<ISessionClient>();
-	const chatMsgOff = chat.notifications.on("msg", (client, _message) => {
-		if (!chatClients.has(client)) {
-			console.log(`client ${client.sessionId} has started chatting`);
-			chatClients.add(client);
-		}
-	});
-	chatMsgOff();
+			// We are configuring the runtime to be in a connected state, so ensure it looks connected
+			runtime.connected = true;
+
+			clock.setSystemTime(initialTime);
+
+			// Set up the presence connection
+			presence = prepareConnectedPresence(runtime, "sessionId-2", "client2", clock, logger);
+
+			// Get a notifications workspace
+			notificationsWorkspace = presence.getNotifications("name:testNotificationWorkspace", {});
+		});
+
+		afterEach(function (done: Mocha.Done) {
+			clock.reset();
+
+			// If the test passed so far, check final expectations.
+			if (this.currentTest?.state === "passed") {
+				assertFinalExpectations(runtime, logger);
+			}
+			done();
+		});
+
+		after(() => {
+			clock.restore();
+		});
+
+		it("can be created via `Notifications` added to workspace", async () => {
+			// Act
+			notificationsWorkspace.add(
+				"testEvents",
+				Notifications<
+					// Below explicit generic specification should not be required.
+					{
+						newId: (id: number) => void;
+					},
+					"testEvents"
+				>({
+					newId: (_client: ISessionClient, _id: number) => {},
```

Comment:
Let's also test that an initial subscription is not required by omitting one here.
It looks like `Notifications` function will need to update argument to be partial:
```
	initialSubscriptions: Partial<NotificationSubscriptions<T>>,
```